_posts/2022-02-15-label-noise-intro.md

```diff
@@ -0,0 +1,131 @@
+| overlapping_intent               | Intent definitions are not mutually exclusive.                                                                                                                                                |
+| renamed_intent                   | Intent renamed after guideline changes. This is not exactly a tagging error, but it’s helpful to capture changes.                                                                             |
+| missing_context                  | Impossible to understand the intent unless more information is provided (bot prompt or previous state etc.)                                                                                   |
+| wrong_retag                      | The new label after re-tagging is wrong. This is not a cause but it allows to capture confusing intents after rechecks.                                                                                                                                                      |
+* since our intent classifiers were not multi-label, we wanted to capture the total % of multiple intent scenarios.\
+We observed that the label noise patterns for each client were quite different. [hard to build a generalizable model where there not enough correct instances of the instance in the first place]
+
+
+## Different cleaning methods to fix the label noise
+
+By measuring the reduction in cleaning effort from the baseline, we can assess the efficacy of the cleaning method. We plotted label noise recall vs % of samples retagged (or annotator effort) - Relating these metrics with previous impact graph allows us to reach interesting conclusions like - *clean y% of the dataset using a method M, and you will get some x% bump in model performance*. 
+
+### Random Sampling
+Here we can sample some fixed number of instances and get them retagged. This serves as our baseline for other methods. The label noise we capture will be around 13% of each partial sample, and hence the recall will be the fraction of the partial sample (in the whole). On average, the plot will look similar to y=x, like this one for our dataset:
+
+![image info](../assets/images/label-noise-blog/random-sampling-plot.png)
+
+### Biased Sampling
+This requires intermittent involvement from ops (tagging after every sampling iteration)
+
+In this method, we first randomly retag x % of samples. Then we identify the major tag confusions (as shown in the Dataset section) and pick the top 5 noisy tags and increase the weights associated to these tags in the sampling function. Then we sample again - pick top 5 tags - repeat.
+
+![image info](../assets/images/label-noise-blog/random-sampling-weights-plot.png)
+
+We see an improvement over the baseline. We can capture around 60% of the total label noise by just tagging around 32% of the total dataset by this heuristic.
+
+### Datamaps
```

Comment:
while reading, this digression is breaking the flow. maybe we can put it as an appendix ?